android/src/main/java/com/joinflux/flux/capacitorsegment/CapacitorSegmentPlugin.java

```diff
@@ -26,24 +26,24 @@ public void initialize(PluginCall call) {
             return;
         }
         String key = call.getString("key");
-        Boolean trackLifecycle = call.getBoolean("trackLifecycle", false);
         if (key == null) {
             call.reject("Write key is required to initialize plugin");
             return;
         }
 
         Context context = this.getContext();
-        this.getActivity().runOnUiThread(new Runnable() {
-            @Override
-            public void run() {
-                Builder builder = new Analytics.Builder(context, key);
-                if (trackLifecycle) {
-                    builder.trackApplicationLifecycleEvents();
-                }
-                initialized = true;
-                implementation.analytics = builder.build();
-            }
-        });
+        Builder builder = new Analytics.Builder(context, key);
+        Boolean trackLifecycle = call.getBoolean("trackLifecycle", false);
+        if (trackLifecycle) {
+            builder.trackApplicationLifecycleEvents();
+        }
+
+        Boolean recordScreenViews = call.getBoolean("recordScreenViews", false);
+        if (recordScreenViews) {
+            builder.recordScreenViews();
```

Comment:
Reading https://segment.com/docs/connections/sources/catalog/libraries/mobile/android/#screen, does this let us basically track the user's entire app journey?

If so, are we risking pushing too many events to Segment?

Reply:
I believe "record" in those docs mean "track" and not actually record the screen. We already track "screens" but since we use capacitor we don't leverage native screens just route changes.

Reply:
Right, yes, I also meant track the "page names/paths" that they navigate to (not video screen-recording). If a user navigates to 10 different "screens", that's 10 different events, yeah?

And I see, if we're tracking that now, then I guess it's probably not a problem (volume-wise)

Reply:
Just FYI I've added this for completeness. I don't think we will be leveraging it in the short term.